### PR TITLE
Fix some bugs in idl

### PR DESCRIPTION
**Pull request by huangd0ng**
Default values of scalar type were not parsed. For example:
```
query(arg : Int = 100) : String
```
default value of arg should be Integer but BigInteger is got.

Directives are not printed in SchemaPrinter currently.

### CODE REVIEW

File: src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy [hunk 1]
```diff
@@ -2,21 +2,7 @@ package graphql.schema.idl
...
-import graphql.schema.GraphQLInterfaceType
-import graphql.schema.GraphQLList
-import graphql.schema.GraphQLNonNull
-import graphql.schema.GraphQLObjectType
-import graphql.schema.GraphQLScalarType
-import graphql.schema.GraphQLSchema
-import graphql.schema.GraphQLType
-import graphql.schema.GraphQLUnionType
-import graphql.schema.TypeResolver
+import graphql.schema.*
```
Comment (andimarek) [inline src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy]: we don't use wildcard imports: please change that to the previous style.

File: src/main/java/graphql/schema/idl/SchemaPrinter.java [hunk 2]
```diff
@@ -399,4 +462,56 @@ private String getDescription(Object descriptionHolder) {
             return Assert.assertShouldNeverHappen();
         }
     }
+
+    private List<Directive> getDirectives(Object directiveHolder) {
```
Comment (andimarek) [inline src/main/java/graphql/schema/idl/SchemaPrinter.java]: is this method a copy from somewhere? If not we should add more tests for the different cases

**Comment by andimarek** [comment]
@huangd0ng Thanks a lot for the PR! I did a review and found some issues, but I am sure we can resolve them.

**Comment by huangd0ng** [comment]
@andimarek I saw your comments and I'll submit a patch after work. We are using this library in production and feel good. Thanks for your work!

File: src/main/java/graphql/schema/idl/SchemaPrinter.java (hunk 2 above)
Comment (bbakerman) [inline src/main/java/graphql/schema/idl/SchemaPrinter.java]: The tests should cover this

File: src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy [hunk 3]
```diff
@@ -331,6 +317,24 @@ type Query {
...
+        def queryType = GraphQLObjectType.newObject().name("Query").field(fieldDefinition).build()
+        def schema = GraphQLSchema.newSchema().query(queryType).build()
+        when:
+        def result = new SchemaPrinter().print(schema)
+
+        then:
+        result == """type Query {
+  field: String @deprecate(reason: "reason")
+}
+"""
```
Comment (bbakerman) [inline src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy]: Yeah the test coverage here is not wide enough.  It does not cover the 7 different types of objects

**Comment by bbakerman** [comment]
I took the SchemGen portion of this PR and applied it to master.  The SchemaPrinter code needs to be reimplemented and more extensively tested

**Comment by huangd0ng** [comment]
@bbakerman  @andimarek  i'm sorry that there're too much work to do recently and I haven't made changes according to your reviews. I saw @bbakerman submitted changes based on this. Shall we abandon this PR?

**Comment by bbakerman** [comment]
I think abandon this PR and start a new one with the SchemaPrinter directives fixes separate.

You need to test them more since that was part of the problem with combining 2 changes

I am going to close this and await a new PR

Thanks for contributing to the project.  We do appreciate the support

**Pull request closed without merging**